code/modules/lighting/light_range_icon.dm

```diff
@@ -0,0 +1,168 @@
+// -- Light icon selection.
+
+/proc/try_get_light_range_icon(block_1, block_2, light_range, num)
+	/*
+	//if (lighting_engine)
+	//	lighting_engine.choose_light_range_icon(two_bordering_walls, light_range, num)
+	//else
+	//	world.log << "Auto-init of the lighting engine datum..."
+	//	lighting_engine = new lighting_system_used
+	//	lighting_engine.choose_light_range_icon(two_bordering_walls, light_range, num)
+	*/
+	var/shadowicon
+	switch (num)
```

Comment:
could you not massively compress this file with ternary statements?

Reply:
No, because you can use var reference statements with `'icons/file1.dmi'` and you need to use `file("icons/file[x].dmi"), which is, I kid you not, 10000x times more taxing on the server the files are no longer added to the rsc.